src/eval.c

```diff
@@ -597,7 +597,9 @@ typval2string(typval_T *tv, int convert)
 	ga_append(&ga, NUL);
 	retval = (char_u *)ga.ga_data;
     }
-    else if (convert && tv->v_type == VAR_DICT)
+    else if (tv->v_type == VAR_LIST)
+	retval = list2string(tv, get_copyID(), FALSE);
+    else if (tv->v_type == VAR_DICT)
 	retval = dict2string(tv, get_copyID(), FALSE);
```

Comment:
Maybe this should use `tv2string()` instead? That handles NULL list and NULL dict.

Reply:
Thanks.  Modified the code to use tv2string() for Dict and List.  Add additional tests for NULL List/Dict.